src/tensor_operations/matrix_decomposition.jl

```diff
@@ -360,44 +360,92 @@ function noinds_error_message(decomp::String)
    treating the ITensor as a matrix from the primed to the unprimed indices."
 end
 
+function add_trivial_index(A::ITensor, Ainds)
+  α = trivial_index(Ainds) #If Ainds[1] has no QNs makes Index(1), otherwise Index(QN()=>1)
+  vα = onehot(eltype(A), α => 1)
+  A *= vα
+  return A, vα, [α]
+end
+
+function add_trivial_index(A::ITensor, Linds, Rinds)
+  vαl, vαr = nothing, nothing
+  if isempty(Linds)
+    A, vαl, Linds = add_trivial_index(A, Rinds)
+  end
+  if isempty(Rinds)
+    A, vαr, Rinds = add_trivial_index(A, Linds)
+  end
+  return A, vαl, vαr, Linds, Rinds
+end
+
+function remove_trivial_index(Q::ITensor, R::ITensor, vαl, vαr)
+  if !isnothing(vαl) #should have only dummy + qr,Link
+    Q *= dag(vαl)
+  end
+  if !isnothing(vαr) #should have only dummy + qr,Link
+    R *= dag(vαr)
+  end
+  return Q, R
+end
+
 qr(A::ITensor; kwargs...) = error(noinds_error_message("qr"))
 
-# TODO: write this in terms of combiners and then
-# call qr on the order-2 tensors directly
 function qr(A::ITensor, Linds...; kwargs...)
```

Comment:
Instead of `kwargs...` and then `get`, let's put the keyword arguments directly in the signature with the defaults, i.e.:
```julia
function qr(A::ITensor, Linds...; tags=ts"Link,qr")
  qtag = TagSet(tags)
  # [...]
end
```

Reply:
I like this because it makes it clear what is supported and the defaults right in the function signature. However it is breaking the _factorize()_ function because keys like _:ortho_ and _:which_decomp_ are inadvertently left in the _kwargs_ dict. when _qr()_ is called.  I can modify _factorize()_ to explicitly remove those keys before calling _qr()_ but it looks pretty clunky.  There is also a test in test_svd.jl that inadvertently passes down cutoff=nothing causing another failure.  
Is there way to indicate that un-used named args are allowed?
`